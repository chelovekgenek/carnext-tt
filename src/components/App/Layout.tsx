import React from "react"
import { Link } from "react-router-dom"

import { Header, Logo, Content, ContentWrapper } from "./Layout.styled"

export const Layout: React.FC = ({ children }) => (
  <React.Fragment>
    <Header>
      <Link to="/">
        <Logo src="https://www.carnext.com/static/images/carnext/logo-eu-header.png" alt="logo" />
      </Link>
    </Header>
    <ContentWrapper>
      <Content>{children}</Content>
    </ContentWrapper>
  </React.Fragment>
)
