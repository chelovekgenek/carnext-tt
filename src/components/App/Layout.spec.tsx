import React from "react"
import { shallow } from "enzyme"

import { Layout } from "./Layout"

describe("Layout", () => {
  const render = (props = {}) => shallow(<Layout {...props} />)
  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
  it("should render child as text", () => {
    expect(render({ children: "Hello there!" })).toMatchSnapshot()
  })
  it("should render child as ReactNode", () => {
    expect(render({ children: <div>asd</div> })).toMatchSnapshot()
  })
})
