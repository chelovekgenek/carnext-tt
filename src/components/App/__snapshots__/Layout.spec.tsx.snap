// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<Fragment>
  <styled.header>
    <Link
      to="/"
    >
      <styled.img
        alt="logo"
        src="https://www.carnext.com/static/images/carnext/logo-eu-header.png"
      />
    </Link>
  </styled.header>
  <styled.main>
    <styled.div />
  </styled.main>
</Fragment>
`;

exports[`Layout should render child as ReactNode 1`] = `
<Fragment>
  <styled.header>
    <Link
      to="/"
    >
      <styled.img
        alt="logo"
        src="https://www.carnext.com/static/images/carnext/logo-eu-header.png"
      />
    </Link>
  </styled.header>
  <styled.main>
    <styled.div>
      <div>
        asd
      </div>
    </styled.div>
  </styled.main>
</Fragment>
`;

exports[`Layout should render child as text 1`] = `
<Fragment>
  <styled.header>
    <Link
      to="/"
    >
      <styled.img
        alt="logo"
        src="https://www.carnext.com/static/images/carnext/logo-eu-header.png"
      />
    </Link>
  </styled.header>
  <styled.main>
    <styled.div>
      Hello there!
    </styled.div>
  </styled.main>
</Fragment>
`;
